### PR TITLE
Added viDestroyManagedLayer

### PR DESCRIPTION
**Pull request by averne**

### CODE REVIEW

File: nx/source/services/vi.c [hunk 1]
```diff
@@ -960,7 +992,7 @@ Result viCloseLayer(ViLayer *layer)
         } *resp = r.Raw;
 
         rc = resp->result;
-        memset(layer, 0, sizeof(ViLayer));
+        layer->initialized = false;
```
Comment (fincs) [inline nx/source/services/vi.c]: Why was this changed?

Comment (averne) [reply nx/source/services/vi.c]: To close a managed layer you need to call viCloseLayer and viDestroyManagedLayer successively, but since the layer id was zeroed you'd need to cache it otherwise.

Comment (fincs) [reply nx/source/services/vi.c]: This tells me there's a design problem with the API. I don't think it's reasonable to call a method on an object that has been closed. Probably, viDestroyManagedLayer should be internally called somewhere within viCloseLayer.

Comment (averne) [reply nx/source/services/vi.c]: Ok nevermind, for some reason I remembered calling viDestroyManagedLayer without calling viCloseLayer first would make it error, but after testing it appears that not.
I guess I should zero the layer similar to viCloseLayer then?

Comment (fincs) [reply nx/source/services/vi.c]: Sure.
